presto-main/src/main/java/com/facebook/presto/sql/planner/SqlPlannerContext.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.SystemSessionProperties;
+import com.facebook.presto.spi.PrestoException;
+
+import static com.facebook.presto.spi.StandardErrorCode.EXCEEDED_PLAN_NODE_LIMIT;
+import static java.lang.String.format;
+
+public class SqlPlannerContext
+{
+    // Record the current number of scan nodes in query plan during planning
+    private int scanNodeInLogicalPlan;
+
+    public SqlPlannerContext(int scanNodeInLogicalPlan)
+    {
+        this.scanNodeInLogicalPlan = scanNodeInLogicalPlan;
+    }
+
+    public void incrementPlanNodes(Session session)
+    {
+        scanNodeInLogicalPlan += 1;
+        if (SystemSessionProperties.isPlanNodeLimitEnabled(session)) {
```

Comment:
import static all calls of `SystemSessionProperties.function(session)` and use them. We generally prefer import static calls if function names are specific enough

Reply:
Done.